core/src/main/scala/org/http4s/MultipartParser.scala

```diff
@@ -0,0 +1,119 @@
+package org.http4s
+
+import scala.language.postfixOps
+
+import scodec.bits.ByteVector
+
+import scalaz.{-\/, \/-, \/}
+import scalaz.stream.{process1, Process, Process1, Writer1}
+
+import org.http4s.util.string._
+
+object MultipartParser {
+  import Process._
+
+  def parse: Writer1[Headers, ByteVector, ByteVector] = {
+    def receiveLine(leading: Option[ByteVector]): Writer1[ByteVector, ByteVector, (ByteVector, Option[ByteVector])] = {
+      def handle(bv: ByteVector): Writer1[ByteVector, ByteVector, (ByteVector, Option[ByteVector])] = {
+        val index = bv indexOfSlice ByteVector('\r', '\n')
```

Comment:
What happens here if we we happen to get two chunks as such: ["first\r","\n--TOKEN"]? It looks to me like this will fail to split the line and then all is lost.